ui/src/Signin.tsx

```diff
@@ -74,13 +75,28 @@ export class Signin extends PureComponent<Props, State> {
   private checkForLogin = async () => {
     try {
       await client.users.me()
+      const redirectIsSet = !!window.localStorage.getItem('redirectTo')
+      if (redirectIsSet) {
+        window.localStorage.removeItem('redirectTo')
+      }
     } catch (error) {
       const {
         location: {pathname},
       } = this.props
 
       clearInterval(this.intervalID)
 
+      if (CLOUD && isFlagEnabled('redirectto')) {
+        const url = new URL(
+          `${window.location.origin}${CLOUD_SIGNIN_PATHNAME}?redirectTo=${
+            window.location.href
+          }`
+        )
+        window.localStorage.setItem('redirectTo', window.location.href)
```

Comment:
note to self: this is because sessions are handled weird on the backend and don't exist until a user is authenticated instead of elevating an anonymous session that is created on first touch. We can't follow a request chain from the unauthenticated user to an authenticated user. this'll become a problem when we start sharing dashboards as we'll introduce non-authenticated user state into the mix.